docs/decisions/0014-access-policy-grouping.rst

```diff
@@ -0,0 +1,40 @@
+0014 Access Policy Grouping
+###########################
+
+Status
+******
+**Pending** Feb 2024
+
+Context
+*******
+The enterprise-access service needs to allow for the flexibility to control and distribute subsidy access policies to
+custom subsets of learners. The subdivision of access policies will improve user management and budgeting from an admin
+perspective, as well as add support for personalization to the learner experience down the line. While it is not
+the responsibility of the service to house the grouping of learners, an access policy must take into account the 
+existence of related subsets when determining if a subsidy is redeemable by an individual learner.
+
+Decision
+********
+The ``SubsidyAccessPolicy`` model will adopt a new, nullable UUID field: `enterprise_group_uuid`. This field will be
```

Comment:
Do we want this to be a nullable field? With this change, by default, every single policy will have a group, and all the learners will be under this group.

Reply:
my thought was backwards compatibility- this way we can have a chance to at a minimum backfill all the existing policies with new groups

Reply:
+ there's some order of operations stuff going on, ie we need to create the policy then the group then update the policy with the group uuid

Reply:
Yeah that makes sense, but I also think we should call out that it really shouldn't be null in practice

Reply:
Sure, I made an addition

Reply:
yeah i think for backwards compatibility we'll want this to be nullable. We can also discuss different ways to implement the "all users in the enterprise" group which is an implicit group today. One approach is the "blank" or "no group specified" means "all users in the enterprise". Alternatively you can have a magic/dynamic group which is all users (if you feel the need to be explicit with a UUID for "all users" of a given enterprise customer).

Reply:
Agreed and I feel that leaving the field nullable keeps us open to either implementation